internal/security/bootstrapper/mosquitto/handlers/handlers.go

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License
+* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing permissions and limitations under
+* the License.
+*
+*******************************************************************************/
+
+package handlers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/mosquitto/container"
+
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/v2/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+)
+
+const (
+	// file read/write permission only for owner
+	readWriteOnlyForOwner os.FileMode = 0600
+)
+
+// Handler is the redis bootstrapping handler
+type Handler struct {
+	credentials bootstrapConfig.Credentials
+}
+
+// NewHandler instantiates a new Handler
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+// GetCredentials retrieves the mosquitto credentials from secretstore
+func (handler *Handler) GetCredentials(ctx context.Context, _ *sync.WaitGroup, startupTimer startup.Timer,
+	dic *di.Container) bool {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	config := container.ConfigurationFrom(dic.Get)
+	secretProvider := bootstrapContainer.SecretProviderFrom(dic.Get)
+
+	var credentials *bootstrapConfig.Credentials
+
+	for startupTimer.HasNotElapsed() {
+		// retrieve mosquitto credentials from secretstore
+		secrets, err := secretProvider.GetSecret(config.MessageQueue.SecretName)
+		if err == nil {
+			credentials = &bootstrapConfig.Credentials{
+				Username: secrets[secret.UsernameKey],
+				Password: secrets[secret.PasswordKey],
+			}
+			break
+		}
+
+		lc.Warnf("Could not retrieve mosquitto credentials (startup timer has not expired): %s", err.Error())
+		startupTimer.SleepForInterval()
+	}
+
+	if credentials == nil {
+		lc.Error("Failed to retrieve mosquitto credentials before startup timer expired")
+		return false
+	}
+
+	handler.credentials = *credentials
+	return true
+}
+
+// SetupPasswordFile sets up the mosquitto password file
```

Comment:
This one is very specific to `mosquitto` so I am good with it here in the `mosquitto/handlers/` folder/file